cpp/src/io/parquet/parquet.hpp

```diff
@@ -45,6 +45,104 @@ struct file_ender_s {
   uint32_t magic;
 };
 
+// thrift generated code simplified.
+struct StringType {
+};
+struct MapType {
+};
+struct ListType {
+};
+struct EnumType {
+};
+struct DecimalType {
+  int32_t scale     = 0;
+  int32_t precision = 0;
+};
+struct DateType {
+};
+
+struct MilliSeconds {
+};
+struct MicroSeconds {
+};
+typedef struct _TimeUnit__isset {
+  _TimeUnit__isset() : MILLIS(false), MICROS(false) {}
+  bool MILLIS;
+  bool MICROS;
+} _TimeUnit__isset;
+
+struct TimeUnit {
+  _TimeUnit__isset __isset;
+  MilliSeconds MILLIS;
+  MicroSeconds MICROS;
+};
+
+struct TimeType {
+  bool isAdjustedToUTC = false;
+  TimeUnit unit;
+};
+struct TimestampType {
+  bool isAdjustedToUTC = false;
+  TimeUnit unit;
+};
+struct IntType {
+  int8_t bitWidth = 0;
+  bool isSigned   = false;
+};
+struct NullType {
+};
+struct JsonType {
+};
+struct BsonType {
+};
+
+// thrift generated code simplified.
+typedef struct _LogicalType__isset {
+  _LogicalType__isset()
+    : STRING(false),
+      MAP(false),
+      LIST(false),
+      ENUM(false),
+      DECIMAL(false),
+      DATE(false),
+      TIME(false),
+      TIMESTAMP(false),
+      INTEGER(false),
+      UNKNOWN(false),
+      JSON(false),
+      BSON(false)
+  {
+  }
+  bool STRING;
+  bool MAP;
+  bool LIST;
+  bool ENUM;
+  bool DECIMAL;
+  bool DATE;
+  bool TIME;
+  bool TIMESTAMP;
+  bool INTEGER;
+  bool UNKNOWN;
+  bool JSON;
+  bool BSON;
+} _LogicalType__isset;
```

Comment:
Can more than one bool be set to true in this struct?

Reply:
parquet logical types say it's a union.
`LogicalType. The new representation is a union of struct of logical types`
When I generated the cpp code from [parquet.thrift](https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift), it generated code which allowed it to be set simultaneously. So, I adopted similar strategy here. 
In future, it would be nice to generate all these structs, with parquet.thrift.

Reply:
parquet.thrift generated code makes it look like you can have more than one bit set for LogicalType. This is because LogicalType::__set_<TYPE>(...) functions set a bit associated with <TYPE> to true but don't unset everything else.

If that was not the case then all of these bools could have been replaced by a single enum.